pkg/cmd/kubeconfig.go

```diff
@@ -45,21 +53,38 @@ func kubeconfigCmd(rootFlags *pflag.FlagSet) *cobra.Command {
 				return err
 			}
 
-			return runKubeconfig(gopts)
+			opts.globalOptions = *gopts
+			st, err := opts.BuildState()
+			if err != nil {
+				return err
+			}
+
+			return runKubeconfig(st, opts.SuperAdmin)
 		},
 	}
 
+	cmd.Flags().BoolVarP(
+		&opts.SuperAdmin,
+		longFlagName(opts, "SuperAdmin"),
+		shortFlagName(opts, "SuperAdmin"),
+		false,
+		"generate short-lived system:masters kubeconfig")
+
 	return cmd
 }
 
 // runKubeconfig downloads kubeconfig file
-func runKubeconfig(opts *globalOptions) error {
-	s, err := opts.BuildState()
-	if err != nil {
-		return err
-	}
+func runKubeconfig(st *state.State, generateSuperAdmin bool) error {
+	var (
+		konfig []byte
+		err    error
+	)
 
-	konfig, err := kubeconfig.Download(s)
+	if generateSuperAdmin {
+		konfig, err = kubeconfig.GenerateSuperAdmin(st)
+	} else {
+		konfig, err = kubeconfig.Download(st)
```

Comment:
I'm wondering if we should replace this with generating our own kubeconfig where users can specify the group and for how long the kubeconfig should be valid for.